samples/algorithms/Teleportation.qs

```diff
@@ -33,13 +33,13 @@ namespace Sample {
             // function.
             initializer(message);
             Message($"Teleporting state {state}");
-            DumpMachine();
```

Comment:
Thanks for the changes @goshua13!

One thing that we @swernli and I noticed is that `DumpRegister` output for teleportation of state `|-〉` because it shows a global phase.

In order to reduce global phase confusion, could you please also make the following changes to this sample:
- Move the allocation of the `message` and `target` qubits inside the loop. This will deallocate them in every iteration and will remove any global phase accumulated from previous teleportations because all qubits are released for every iteration.
- Inside the `Teleport` operation, switch the ordering of the corrections. The current order introduces a global phase that makes things a bit more confusing.

Reply:
Does that help? I tried that but still got -1 phase on |-> half of the times

Reply:
It helps for the global state. With these changes the output of `DumpMachine` is consistent. However, as part of looking into this, we found a potential bug in `DumpRegister` and that's why sometimes we get a -1 phase and sometimes not. We specifically get the -1 phase when we apply a `Z` correction.

Reply:
Yup, like Cesar said these changes make the internal state more consistent, which only highlights how odd it is that `DumpRegister` specifically adds an unexpected phase.

We tried this out with just the minus state and printing what corrections are applied. With the updates Cesar suggested, we can see that when no corrections are applied the before and after of both `DumpMachine` (shown first) and `DumpRegister` (shown second) agree:
<img width="571" alt="image" src="https://github.com/microsoft/qsharp/assets/10567287/4a213745-6625-46af-b058-7ede51b2d3c5">

But when a Z correction is applied, the message qubit is left in the one state, and somehow that causes `DumpRegister` to report the target qubit as having a phase of -1:
<img width="571" alt="image" src="https://github.com/microsoft/qsharp/assets/10567287/4939c298-b7dd-4ebf-89d8-a7de45983cca">

Given that the underlying amplitudes are the same, the "other" qubit being in the one state should not cause `DumpRegister` on the target qubit to include a phase.